python/predict.py

```diff
@@ -57,18 +59,30 @@ def magentic_data_callback(data):
     x_test = x_test.reshape((1, len(x_test)))
     x_test = sensors_scaler[data.id].transform(x_test).astype('float32')
 
-    # Pass data into neural network and output sin and cos, then convert them to Euler Angles
-    output = model[data.id].predict(x_test)
-    output = sin_cos_to_angle(output)
+    if use_lstm[data.id]:
+
+        history[data.id] = np.append(history[data.id], x_test, axis=0)
+        if len(history[data.id]) <= history_n:
+            return
+        else:
+            history[data.id] = np.delete(history[data.id], 0, axis=0)
+
+        # Pass data into neural network and output sin and cos, then convert them to Euler Angles
+        output = model[data.id].predict(history[data.id][None, :, :])
+        output = sin_cos_to_angle(output)
+    else:
+        # Pass data into neural network and output sin and cos, then convert them to Euler Angles
+        output = model[data.id].predict(x_test)
+        output = sin_cos_to_angle(output)
 
     if len(filter[data.id]) < filter_n:
         filter[data.id] = np.append(filter[data.id], output, axis=0)
-        return
+        # return
 
-    avg = np.mean(filter[data.id], axis=0)
-    error = np.abs(avg - output).sum()
+    # avg = np.mean(filter[data.id], axis=0)
+    error = np.abs(filter[data.id][-1] - output).max()
 
-    if error < 0.5:
+    if error < 0.15:
```

Comment:
let's try to avoid magic numbers in the code, so define 0.15 in the beginning of the file with something like `MAX_ERROR`